### ISSUE TITLE
Finalize the general ML use case 

### ISSUE THREAD

**Issue by ChatSam** [ChatSam/Weapons_detection_immich#2]
Finalize which machine learning use case to implement between Violence Detection  and Threat Detection.

To accomplish that, address the following questions - 
1. Which version benefits more from Depth Estimation 
2. Which one has more learning potential?
3. Which one has more use for Child Protection? 
4. Which one is more technically challenging to implement 
5. Which one has more data to train the ML model?

**Comment by ChatSam** [comment]
1. It seems that depth estimation and violence detection don't seem to be combined in any projects 
2. The weapon detection use case seems to be technically more straightforward.

Sources to confirm findings 


1. [MiDaS and YOLO](https://www.notion.so/Violence-Detection-Ideas-2ca088a9055147db92f9ad2bbc5813e3?pvs=21)
2. [Combining YOLO and MiDaS](https://github.com/ultralytics/ultralytics/issues/3653)
3. [Paper - Does depth estimation help object detection ? ](https://arxiv.org/pdf/2204.06512.pdf)
4. [Getting started with MiDaS](https://medium.com/artificialis/getting-started-with-depth-estimation-using-midas-and-python-d0119bfe1159)
5. [Using MiDaS](https://mpolinowski.github.io/docs/IoT-and-Machine-Learning/ML/2023-01-08-depth-vision-midas/2023-01-08/)
6. [Yolo V8 vs V4](https://roboflow.com/compare/yolov8-vs-yolov4)

In Conclusion, weapons detection seems to be a better use case to pursue depth estimation
